app/models/user.rb

```diff
@@ -11,32 +8,28 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
   has_many :friendships
   has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
 
-  def friends
-    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
-    friends_array + inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
-    friends_array.compact
-  end
+  has_many :confirmed_friendships, -> { where confirmed: true }, class_name: 'Friendship'
+  has_many :pending_friendships, -> { where confirmed: false }, class_name: 'Friendship'
+  has_many :friends, through: :confirmed_friendships
 
-  # def non_friends
-  #   friends_array = friendships.map { |friendship| friendship.friend if !friendship.status == 'confirmed' }
-  #   friends_array + inverse_friendships.map { |friendship| friendship.user if !friendship.status == 'confirmed' }
+  # def friends
+  #   friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
+  #   friends_array + inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
   #   friends_array.compact
   # end
```

Comment:
- [ ] Kindly remove commented code in your project. This will keep the code clean, remember you can always use git history to go back to previous versions of the code.